underpants.go

```diff
@@ -111,6 +127,7 @@ func (u *user) encode(key []byte) (string, error) {
   if err := json.NewEncoder(w).Encode(u); err != nil {
     return "", err
   }
+  w.Close()
```

Comment:
No worries, I'm kinda stumbling around in Go so it's good to think it through :)